modules/apps/transfer/keeper/relay_test.go

```diff
@@ -149,16 +157,92 @@ func (suite *KeeperTestSuite) TestSendTransfer() {
 
+func (suite *KeeperTestSuite) TestSendTransferDoesNotSetTotalEscrowAmountForSourceIBCToken() {
+	/*
+		Given the following flow of tokens:
+
+		chain A (channel 0) -> (channel-0) chain B (channel-1) -> (channel-1) chain A
+		stake                  transfer/channel-0/stake           transfer/channel-1/transfer/channel-0/stake
+		                                  ^
+		                                  |
+		                             SendTransfer
+
+		This test will transfer vouchers of denom "transfer/channel-0/stake" from chain B
+		to chain A over channel-1 to assert that no total escrow amount is stored on chain B.
+
+		Set up:
+		- Two transfer channels between chain A and chain B.
+		- Tokens of native denom "stake" on chain A transferred to chain B over channel-0
+		and vouchers minted with denom trace "tranfer/channel-0/stake".
+
+		Execute:
+		- Transfer vouchers of denom trace "tranfer/channel-0/stake" from chain B to chain A
+		over channel-1.
+
+		Assert:
+		- Since the vouchers are not of a native denom, but chain B is the source, then no
+		value for total escrow amount should be stored for denom trace "tranfer/channel-0/stake".
+	*/
+
+	// set up
+	// 2 transfer channels between chain A and chain B
+	path1 := NewTransferPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path1)
+
+	path2 := NewTransferPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path2)
+
+	// create IBC token on chain B with denom trace "transfer/channel-0/stake"
+	coin := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
+	transferMsg := types.NewMsgTransfer(
+		path1.EndpointA.ChannelConfig.PortID,
+		path1.EndpointA.ChannelID,
+		coin,
+		suite.chainA.SenderAccount.GetAddress().String(),
+		suite.chainB.SenderAccount.GetAddress().String(),
+		suite.chainB.GetTimeoutHeight(), 0, "",
+	)
+	result, err := suite.chainA.SendMsgs(transferMsg)
+	suite.Require().NoError(err) // message committed
+
+	packet, err := ibctesting.ParsePacketFromEvents(result.GetEvents())
+	suite.Require().NoError(err)
+
+	err = path1.RelayPacket(packet)
+	suite.Require().NoError(err)
+
+	// execute
+	trace := types.ParseDenomTrace(types.GetPrefixedDenom(path1.EndpointB.ChannelConfig.PortID, path1.EndpointB.ChannelID, sdk.DefaultBondDenom))
+	coin = sdk.NewCoin(trace.IBCDenom(), sdk.NewInt(100))
+	msg := types.NewMsgTransfer(
+		path2.EndpointB.ChannelConfig.PortID,
+		path2.EndpointB.ChannelID,
+		coin,
+		suite.chainB.SenderAccount.GetAddress().String(),
+		suite.chainA.SenderAccount.GetAddress().String(),
+		suite.chainA.GetTimeoutHeight(), 0, "",
+	)
+
+	res, err := suite.chainB.GetSimApp().TransferKeeper.Transfer(sdk.WrapSDKContext(suite.chainB.GetContext()), msg)
+	suite.Require().NoError(err)
+	suite.Require().NotNil(res)
+
+	// check total amount in escrow of sent token on sending chain
+	totalEscrow := suite.chainB.GetSimApp().TransferKeeper.GetTotalEscrowForDenom(suite.chainB.GetContext(), coin.GetDenom())
+	suite.Require().Equal(math.ZeroInt(), totalEscrow)
```

Comment:
Can we check chainA totalEscrow and see if it's 100stake?

Reply:
Yes, I think we could that check here as well, although that scenario should have been checked in the test `successful transfer with native token` of `TestSendTransfer` above.